### PR TITLE
DNN-19423

### PR DESCRIPTION
**Pull request by mean2me**

### CODE REVIEW

File: src/Modules/Settings/Dnn.PersonaBar.SiteSettings/SiteSettings.Web/src/components/editLanguagePanel/resourceEditor/index.jsx [hunk 1]
```diff
@@ -87,9 +87,7 @@ class ResourceEditor extends Component {
             inFullMode: false
         });
 
-        setTimeout(() => {
-            window.dnn.stopEscapeFromClosingPB = false;
-        }, 500);
+        window.dnn.stopEscapeFromClosingPB = false;
```
Comment (tpluscode) [inline src/Modules/Settings/Dnn.PersonaBar.SiteSettings/SiteSettings.Web/src/components/editLanguagePanel/resourceEditor/index.jsx:90]: Every other `setTimeout` was wrapping access to a `this`. Here is the only case where the callback accesses the global scope. Isn't this one actually safe?

File: src/Modules/Settings/Dnn.PersonaBar.SiteSettings/SiteSettings.Web/src/components/languageSettings/languageVerifier/languageVerifierGrid/index.jsx [hunk 2]
```diff
@@ -14,11 +14,9 @@ class LanguageVerifierGridPanel extends Component {
     }
 
     uncollapse(id) {
-        setTimeout(() => {
-            this.setState({
-                openId: id
-            });
-        }, this.timeout);
```
Comment (tpluscode) [inline src/Modules/Settings/Dnn.PersonaBar.SiteSettings/SiteSettings.Web/src/components/languageSettings/languageVerifier/languageVerifierGrid/index.jsx:21]: If you remove `setTimeout` I'd assume that `this.timeout` would also be unnecessary. Where does it come from? I only ever see it being read.

**Review by tpluscode — COMMENTED** [review]
I searched quickly and didn't immediately find a resource discouraging `setTimeout`within React components. Could you post some?

Still, this code was written this way for a reason. When you remove these delays, won't it change the behaviour of the application?

File: src/Modules/Settings/Dnn.PersonaBar.SiteSettings/SiteSettings.Web/src/components/languageSettings/languageVerifier/languageVerifierGrid/index.jsx (hunk 2 above)
Comment (mean2me) [reply src/Modules/Settings/Dnn.PersonaBar.SiteSettings/SiteSettings.Web/src/components/languageSettings/languageVerifier/languageVerifierGrid/index.jsx:21]: Looks like it was undefined.

File: src/Modules/Settings/Dnn.PersonaBar.SiteSettings/SiteSettings.Web/src/components/editLanguagePanel/resourceEditor/index.jsx (hunk 1 above)
Comment (mean2me) [reply src/Modules/Settings/Dnn.PersonaBar.SiteSettings/SiteSettings.Web/src/components/editLanguagePanel/resourceEditor/index.jsx:90]: it is useless to wrap a variable assignment like this into a setTimeout.

File: src/Modules/Settings/Dnn.PersonaBar.SiteSettings/SiteSettings.Web/src/components/languageSettings/languageVerifier/languageVerifierGrid/index.jsx (hunk 2 above)
Comment (mean2me) [reply src/Modules/Settings/Dnn.PersonaBar.SiteSettings/SiteSettings.Web/src/components/languageSettings/languageVerifier/languageVerifierGrid/index.jsx:21]: IMO, undefined

**Comment by mean2me** [comment]
After investigating and testing this application I found that setTimeouts are not used for a specific design purpose. They are like manually adjusted in a "slow enough to work" way. 
All starts from the asynchronous way the extension module is loaded. After I fixed all, I couldn't see any regression.

**Comment by tpluscode** [comment]
fixes #559

**Pull request merged**